server/functions/index.js

```diff
@@ -263,25 +303,33 @@ exports.expireOldFavorsOnCreate = functions.firestore
 // send new favor notification to users on updates
 exports.sendNotificationOnNewChat = functions.firestore
     .document('/chats/{chatId}')
-    .onCreate((snap, context) => {
-        const isFirstMsg = snap.data().isFirstMsg;
-        if (isFirstMsg === "true") {
-            const titleToSend = "You've got a new message";
-            const favorId = snap.data().favorId;
-            const receivers = [snap.data().notifId];
-            const message = {
-                data: {
-                    FavorId: favorId,
-                },
-                notification: {
-                    title: titleToSend,
-                    body: 'Click to check out the chat message',
-                },
-                tokens: receivers
-            };
-            return sendMulticastMessage(message, receivers);
+    .onCreate((snap) => {
+        const message = snap.data();
+        const senderId = message.uid;
+        const favorId = message.favorId;
 
-        }
+        // find favor with selected favorId
+        return db.collection('/favors').where('id', '==', favorId).get()
+            .then((snapshot) => {
+                snapshot.forEach((doc) => {
```

Comment:
would a foreach loop here be appropriate given that we're only fetching 1?

Reply:
Yeah right, this is also true for the other functions we have (e.g. when retrieving a user based on the id). However I didn't find any other ways to handle query results in the documentation or in other forums (maybe I didn't see or find it). I think this is safe to do and probably not inefficient. However, if anyone knows how to handle this in code in a better way, feel free to suggest (sorry but I don't know Javascript very well).

Reply:
If no other ways it's not a problem